projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,58 +1,77 @@
 
-export interface SelectControl {
-  placeholder?: string;
-  default?: SelectOption<TableControlOption>;
-  options: SelectOption<TableControlOption>[];
+export const enum TableControlOptionType {
+  Filter = 'filter',
+  Property = 'property',
+  Unset = 'unset'
 }
 
-export interface SelectChange {
-  select: SelectControl;
-  value: TableControlOption;
-}
+export type TableControlOption = TableUnsetControlOption | TableFilterControlOption | TablePropertyControlOption;
 
-export interface CheckboxControl {
+export interface TableFilterControlOption {
+  type: TableControlOptionType.Filter;
   label: string;
-  value: boolean;
-  options: TableCheckboxOptions;
+  metaValue: TableFilter;
 }
 
-export interface CheckboxChange {
-  checkbox: CheckboxControl;
-  option: TableControlOption<boolean>;
+export interface TableUnsetControlOption {
+  type: TableControlOptionType.Unset;
+  label: string;
+  metaValue: string;
 }
 
-export const enum TableControlOptionType {
-  Filter = 'filter',
-  Property = 'property',
-  UnsetFilter = 'unset-filter'
+export interface TablePropertyControlOption {
+  type: TableControlOptionType.Property;
+  label: string;
+  metaValue: Dictionary<unknown>;
 }
 
-export type TableControlOption<T = unknown> =
-  | TableUnsetFilterControlOption<T>
-  | TableFilterControlOption<T>
-  | TablePropertyControlOption<T>;
+/*
+ * Select Control
+ */
 
-interface TableControlOptionBase<T> {
-  value?: T;
+export interface SelectControl {
+  placeholder: string;
+  options: TableSelectControlOption[];
 }
-export interface TableUnsetFilterControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.UnsetFilter;
-  metaValue: string;
+
+export interface SelectChange {
+  select: SelectControl;
+  values: TableSelectControlOption[];
 }
 
-export interface TableFilterControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.Filter;
-  metaValue: TableFilter;
+export type TableSelectControlOption = TableFilterControlOption;
+
+/*
+ * Checkbox Control
+ */
+
+export interface CheckboxControl {
+  label: string;
+  value: boolean;
+  options: TableCheckboxOptions;
 }
 
-export interface TablePropertyControlOption<T = unknown> extends TableControlOptionBase<T> {
-  type: TableControlOptionType.Property;
-  metaValue: Dictionary<unknown>;
+export interface CheckboxChange {
+  checkbox: CheckboxControl;
+  option: TableCheckboxControlOption;
 }
 
-export type TableCheckboxControlOption<T extends boolean> = TableControlOption<T> & { label: string };
+export type TableCheckboxControlOption<T extends boolean = boolean> = TableControlOption & {
+  value: T;
+};
 
 export type TableCheckboxOptions = [TableCheckboxControlOption<true>, TableCheckboxControlOption<false>];
+
+/*
+ * Util
+ */
+
+export const toInFilter = (tableFilters: TableFilter[]): TableFilter =>
+  tableFilters.reduce((previousValue, currentValue) => ({
+    field: previousValue.field,
```

Comment:
what if the fields in the tableFilter are different?. In this reduce, we are just considering the last field as field for the final In filter.

Reply:
Yeah, this is my issue with this too. 

We could validate here or I can break up the args to ensure we only are passed one `field` and `operator`, but I felt it was overkill and traded a little bit of confusion (and a potential for a bug if used incorrectly) for ease of use. 

I can add some validation easy enough, so I will. I can revisit other options if we feel strongly.